javascripts/discourse/components/custom-header-topic-button.gjs

```diff
@@ -0,0 +1,111 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { action } from "@ember/object";
+import didUpdate from "@ember/render-modifiers/modifiers/did-update";
+import { service } from "@ember/service";
+import DButton from "discourse/components/d-button";
+import I18n from "discourse-i18n";
+
+export default class CustomHeaderTopicButton extends Component {
+  @service composer;
+  @service currentUser;
+  @service router;
+  @service siteSettings;
+
+  @tracked currentRouteAttributes = this.router.currentRoute.attributes;
+  @tracked userHasDraft = this.currentUser.has_topic_draft;
+
+  get currentTag() {
+    return [
+      this.currentRouteAttributes?.tag?.id,
+      ...(this.currentRouteAttributes?.additionalTags ?? []),
+    ]
+      .filter(Boolean)
+      .filter((t) => !["none", "all"].includes(t))
+      .join(",");
+  }
+
+  get canCreateTopic() {
+    return this.currentUser.can_create_topic;
+  }
```

Comment:
This does not need to be a getter and can be a normal property of the component.

```suggestion
  canCreateTopic = this.currentUser.can_create_topic;
```